java/code/src/com/redhat/rhn/taskomatic/task/MinionActionExecutor.java

```diff
@@ -119,27 +121,28 @@ public void execute(JobExecutionContext context) {
         }
 
         // HACK: it is possible that this Taskomatic task triggered before the corresponding Action was really
-        // COMMITted in the database. Wait for some minutes checking if it appears
-        int waitedTime = 0;
-        while (countQueuedServerActions(action) == 0 && waitedTime < ACTION_DATABASE_GRACE_TIME) {
-            if (action != null) {
-                HibernateFactory.getSession().flush();
-                HibernateFactory.getSession().evict(action);
-            }
-
-            action = ActionFactory.lookupById(actionId);
+        // COMMITted in the database. If it is the case, reschedule the Job.
+        int retryCount = (Integer) context.getTrigger().getJobDataMap().getOrDefault("retryCount", 1);
 
-            if (log.isDebugEnabled()) {
-                log.debug("Number of Queued Server Actions for {}: {}", actionId, countQueuedServerActions(action));
-            }
+        if (retryCount <= MAX_RETRIES && countQueuedServerActions(action) == 0) {
+            log.debug("Could not find queued actions, scheduling retry {} of {}", retryCount, MAX_RETRIES);
             try {
-                Thread.sleep(ACTION_DATABASE_POLL_TIME);
+                context.getTrigger().getJobDataMap().put("retryCount", retryCount + 1);
+                Trigger oldTrigger = context.getTrigger();
+                Trigger newTrigger = TriggerBuilder
```

Comment:
Do you think that this retry job creation should be moved to the class `TaskoQuartzHelper`? 
Is the needed anywhere else? What do you think about moving to that class?